transitions/extensions/asyncio.py

```diff
@@ -72,14 +71,11 @@ async def check(self, event_data):
                 model attached to the current machine which is used to invoke
                 the condition.
         """
-        predicate = event_data.machine.resolve_callable(self.func, event_data)
-        if asyncio.iscoroutinefunction(predicate):
```

Comment:
You can still use `asyncio.iscoroutinefunction`.
In this case, it's better.

Reply:
You'd prefer `asyncio.iscoroutinefunction` over `inspect.isawaitable`? Could you eloborate on this? I found [this SO post](https://stackoverflow.com/questions/36076619/test-if-function-or-method-is-normal-or-asynchronous) quite insightful and especially [this answer](https://stackoverflow.com/a/36077430/1617563) gave me the impression that `isawaitable` is the most generic version for that particular use case. The documentation seems to be consistent with claims made in that post as far as I can tell. But I am not attached to `isawaitable` of course. If there are -- for instance performance-wise -- facts that make `asyncio.iscoroutinefunction` more suitable, I am all for it.

Reply:
`asyncio.iscoroutinefunction` returns True for generator-based coroutine functions decorated with @coroutine according to the documentation.

I see no harm in using it.

Reply:
I agree with you that `asyncio.iscoroutinefunction` could be used independently of the chosen event loop. I'd argue however that `inspect.isawaitable` is even better here since callbacks could also be Futures which `asyncio.iscoroutinefunction` is not accepting (see [gist](https://gist.github.com/aleneum/870255aab5094f4dfb9d32408d18cd7f)). For `transitions` it should not matter whether its an async function, a generator or a Future. Eventually, we want to know if we should await something or not.